src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+import duke.task.TaskList;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Handles loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private TaskList list;
+
+    /**
+     * Constructor for Storage class.
+     *
+     * @param list Task list to read/write from the current list of tasks.
+     */
+    public Storage(TaskList list) {
+        this.list = list;
+    }
+
+    /**
+     * Reads from file and store data (tasks) into tasks list.
+     */
+    public void initalizeList() {
+        try {
+            File yourFile = new File("duke.txt");
+            yourFile.createNewFile();
+            try (BufferedReader br = new BufferedReader(new FileReader("duke.txt"))) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    char taskType = line.charAt(1);
+
+                    boolean isMarked = false;
+                    Character markedChar = line.charAt(4);
+
+                    String description = line.substring(7);
+                    if (markedChar.equals('X')) {
+                        isMarked = true;
+                    }
+                    switch (taskType) {
+                        case 'T':
+                            Todo todoTask = new Todo(description, isMarked);
+                            list.add(todoTask);
+                            break;
+                        case 'E':
+                            String modifiedDescription = description.split(" \\(from: ")[1];
+                            LocalDate fromDate = LocalDate.parse(modifiedDescription.split(" to: ")[0], DateTimeFormatter.ofPattern("MMM d yyyy"));
+                            LocalDate toDate = LocalDate.parse(modifiedDescription.split(" to: ")[1].split("\\)")[0], DateTimeFormatter.ofPattern("MMM d yyyy"));
```

Comment:
I believe this line here exceeds the recommended length of 120 characters.

Reply:
Thanks for pointing this out!